src/processor/handler.rs

```diff
@@ -193,21 +194,65 @@ pub trait TransactionContext {
         note = "please use `set_state_entry` or `set_state_entries` instead"
     )]
     /// set_state requests that each address in the provided map be
+    /// set in validator state to its corresponding value. set_state is deprecated, please use
+    /// set_state_entry to set_state_entries instead
+    ///
+    /// # Arguments
+    ///
+    /// * `entries` - entries are a hashmap where the key is an address and value is the data
+    fn set_state(&self, entries: HashMap<String, Vec<u8>>) -> Result<(), ContextError> {
+        self.set_state_entries(entries)
+    }
+
+    /// set_state_entry requests that the provided address is set in the validator state to its
+    /// corresponding value.
+    ///
+    /// # Arguments
+    ///
+    /// * `address` - address of where to store the data
+    /// * `data` - payload is the data to store at the address
+    fn set_state_entry(&self, address: String, data: Vec<u8>) -> Result<(), ContextError>;
```

Comment:
Should address here be &str?

Reply:
Set_state_entry needs to take ownership of the address and the data to set it in the proto message. Also since the handler calling set_state_entry does not need the address anymore (nothing to compare the return type) it is fine to pass on ownership.